spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolver.java

```diff
@@ -32,50 +37,95 @@
 import org.springframework.core.io.Resource;
 import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
 import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.util.StringUtils;
 
 /**
  * Used to retrieve metadata about the configuration properties that can alter an application's behavior.
- *
  * @author Eric Bottard
  */
 public class ApplicationConfigurationMetadataResolver {
 
+	private static final String CONFIGURATION_METADATA_PATTERN = "classpath*:/META-INF/spring-configuration-metadata.json";
+
+	private static final String DATAFLOW_PROPERTIES = "classpath*:/META-INF/dataflow.properties";
+
+	public static final String CONFIGURATION_CLASSES = "configuration.classes";
+
+	public static final String CONFIGURATION_PROPERTIES = "configuration.properties";
+
+	private final Set<String> globalWhiteListedProperties = new HashSet<>();
+
+	private final Set<String> globalWhiteListedClasses = new HashSet<>();
```

Comment:
Hey, thanks for merging this!
The properties file is read from `classpath*` and each file property is merged with the previous. So, to add a global whitelist, we just put a copy of the properties file in dataflow itself (it is read in the resolver constructor using a classloader that "sees" the dataflow classes, as opposed to a classloader that sees an app later). Hope that makes sense